build/cmake/files.cmake

```diff
@@ -588,7 +588,6 @@ set(BASE_CMN_HDR
     wx/textbuf.h
     wx/textfile.h
     wx/thread.h
-    wx/thrimpl.cpp
```

Comment:
`files.cmake` is auto-generated, so the build error caused by this file should be solved in another way.
And preferable in it's own commit.

Reply:
Thanks, I missed the autogenerated note at the top. I'll take a stab at fixing this another way, or remove it from the MR if I can't easily find a fix.

Reply:
Found the root issue here: `set_source_files_properties(... HEADER_FILE_ONLY TRUE)` is used to make the .cpp "headers" not compile. However the `mono` target is created in a different scope so it couldn't read the properties set. I introduced a new variable `wxMONO_NONCOMPILED_CPP_FILES` to pass these upscope so the `mono` target can set this within it's own scope. Fix is pushed.